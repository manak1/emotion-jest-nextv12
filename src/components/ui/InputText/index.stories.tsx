import React from "react"
import { ComponentStory, ComponentMeta } from "@storybook/react"

import InputText from "./index"
import { useArgs } from "@storybook/client-api"

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "InputText",
  component: InputText,
  argTypes: {
    value: { control: "text", defaultValue: "" },
    label: { control: "text", defaultValue: "ラベル" },
    id: { control: "text" },
    onChange: { action: "onChange" },
  },
} as ComponentMeta<typeof InputText>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof InputText> = (args) => {
  const [_, updateArgValues] = useArgs()
  return (
    <InputText {...args} onChange={(value) => updateArgValues({ value })} />
  )
}

export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
