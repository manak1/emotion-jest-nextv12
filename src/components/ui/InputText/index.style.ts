import { css } from "@emotion/react"
import styled from "@emotion/styled"
import { colors, fontSizes } from "~/styles/theme"

/* export const styles = {
  wrapper: css({
    display: "flex",
    flexDirection: "column",
  }),

  input: css({
    marginTop: "4px",
    fontSize: fontSizes.default,
    padding: "8px",
    outline: "none",
    border: "1px solid #d3d3d3",
    borderRadius: "8px",
    "&:focus": {
      borderColor: colors.primary,
    },
  }),
} */

export const Wrapper = styled.div({
  display: "flex",
  flexDirection: "column",
})

export const Input = styled.input(
  {
    marginTop: "4px",
    padding: "8px",
    outline: "none",
    border: "1px solid #d3d3d3",
    borderRadius: "8px",
  },
  ({ theme }) => ({
    fontSize: theme.fontSizes.small,
    "&:focus": {
      borderColor: theme.colors.primary,
    },
  })
)
