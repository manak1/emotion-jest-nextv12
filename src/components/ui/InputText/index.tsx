import React from "react"
import Label from "../Label"

import { styles } from "./index.style"

type Props = React.ComponentProps<"input"> & {
  name: string
  label: string
  required?: boolean
}

const InputText: React.VFC<Props> = (props) => {
  const { name, label, required, ...rest } = props
  return (
    <div css={styles.wrapper}>
      <Label htmlFor={name} label={label} required={required} />
      <input css={styles.input} id={name} type="text" {...rest} />
    </div>
  )
}

export default InputText
