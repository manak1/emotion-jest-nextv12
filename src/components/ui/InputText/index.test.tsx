import { render, screen } from "@testing-library/react"
import InputText from "./index"
import userEvent from "@testing-library/user-event"

describe("InputTextのテスト", () => {
  const onChange = jest.fn()

  it("snapshot", () => {
    const { asFragment } = render(
      <InputText
        value="test"
        label="test"
        id="test"
        placeholder="test"
        onChange={onChange}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("labelが表示される", () => {
    render(
      <InputText
        label="ラベル"
        value=""
        id="test"
        placeholder=""
        onChange={onChange}
      />
    )
    expect(screen.getByLabelText("ラベル")).toBeInTheDocument()
  })

  it("初期値が表示される", () => {
    render(
      <InputText
        label="ラベル"
        value="test"
        id="test"
        placeholder=""
        onChange={onChange}
      />
    )
    expect(screen.getByRole("textbox")).toHaveValue("test")
  })

  it("idが設定されている", () => {
    render(
      <InputText
        label="ラベル"
        value="test"
        id="test"
        placeholder=""
        onChange={onChange}
      />
    )
    expect(screen.getByRole("textbox")).toHaveAttribute("id", "test")
  })

  it("入力時にonChangeが呼ばれる", async () => {
    render(
      <InputText
        label="ラベル"
        value=""
        id="test"
        placeholder=""
        onChange={onChange}
      />
    )
    const input = screen.getByLabelText("ラベル")
    userEvent.type(input, "テスト")
    expect(onChange).toBeCalledTimes(3)
  })
})
