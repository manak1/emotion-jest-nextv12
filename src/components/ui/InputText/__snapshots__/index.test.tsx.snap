// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextのテスト snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-69i1ev"
    >
      <label
        class="css-1lu35ut"
        for="test"
      >
        test
      </label>
    </div>
    <input
      class="css-rknwl4"
      id="test"
      placeholder="test"
      type="text"
      value="test"
    />
  </div>
</DocumentFragment>
`;
